cli/azd/internal/appdetect/pom_test.go

```diff
@@ -1,235 +1,1940 @@
+		t.Run(tt.name, func(t *testing.T) {
+			absorbDependency(&tt.input, tt.toBeAbsorbedPom)
+			if !reflect.DeepEqual(tt.input, tt.expected) {
+				t.Fatalf("\nExpected: %s\nActual:   %s", tt.expected, tt.input)
+			}
+		})
+	}
+}
+
+func TestAbsorbBuildPlugin(t *testing.T) {
+	var tests = []struct {
+		name            string
+		input           pom
+		toBeAbsorbedPom pom
+		expected        pom
+	}{
+		{
+			name: "absorb 2 plugins",
+			input: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactId",
+				Version:    "1.0.0",
+			},
+			toBeAbsorbedPom: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactIdToBeAbsorbed",
+				Version:    "1.0.0",
+				Build: build{
+					Plugins: []plugin{
+						{
+							GroupId:    "groupIdOne",
+							ArtifactId: "artifactIdOne",
+							Version:    "1.0.0",
+						},
+						{
+							GroupId:    "groupIdTwo",
+							ArtifactId: "artifactIdTwo",
+							Version:    "1.0.0",
+						},
+					},
+				},
+			},
+			expected: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactId",
+				Version:    "1.0.0",
+				Build: build{
+					Plugins: []plugin{
+						{
+							GroupId:    "groupIdOne",
+							ArtifactId: "artifactIdOne",
+							Version:    "1.0.0",
+						},
+						{
+							GroupId:    "groupIdTwo",
+							ArtifactId: "artifactIdTwo",
+							Version:    "1.0.0",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "absorb 1 plugin and skip 1 plugin",
+			input: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactId",
+				Version:    "1.0.0",
+				Build: build{
+					Plugins: []plugin{
+						{
+							GroupId:    "groupIdOne",
+							ArtifactId: "artifactIdOne",
+							Version:    "2.0.0",
+						},
+					},
+				},
+			},
+			toBeAbsorbedPom: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactIdToBeAbsorbed",
+				Version:    "1.0.0",
+				Build: build{
+					Plugins: []plugin{
+						{
+							GroupId:    "groupIdOne",
+							ArtifactId: "artifactIdOne",
+							Version:    "1.0.0",
+						},
+						{
+							GroupId:    "groupIdTwo",
+							ArtifactId: "artifactIdTwo",
+							Version:    "1.0.0",
+						},
+					},
+				},
+			},
+			expected: pom{
+				GroupId:    "sampleGroupId",
+				ArtifactId: "sampleArtifactId",
+				Version:    "1.0.0",
+				Build: build{
+					Plugins: []plugin{
+						{
+							GroupId:    "groupIdOne",
+							ArtifactId: "artifactIdOne",
+							Version:    "2.0.0", // keep original value
+						},
+						{
+							GroupId:    "groupIdTwo",
+							ArtifactId: "artifactIdTwo",
+							Version:    "1.0.0",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			absorbBuildPlugin(&tt.input, tt.toBeAbsorbedPom)
+			if !reflect.DeepEqual(tt.input, tt.expected) {
+				t.Fatalf("\nExpected: %s\nActual:   %s", tt.expected, tt.input)
+			}
+		})
+	}
+}
+
+func TestCreateSimulatedEffectivePomFromFilePath(t *testing.T) {
```

Comment:
FYI: When review this PR, if you don't want to review all test codes, just review this one is enough.